File: java-example/src/test/java/ru/stqa/training/selenium/CustomerRegistrationTests.java
package ru.stqa.training.selenium;

import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.Select;

public class CustomerRegistrationTests extends TestBase {

    @Test
    public void canRegisterCustomer() {
        Customer customer = Customer.newEntity()
                .withFirstname("Adam").withLastname("Smith").withPhone("+0123456789")
                .withAddress("Hidden Place").withPostcode("12345").withCity("New City")
                .withCountry("US").withZone("KS")
                .withEmail("adam"+System.currentTimeMillis()+"@smith.me")
                .withPassword("qwerty").build();

        driver.get("http://localhost/litecart/en/create_account");
        driver.findElement(By.name("firstname")).sendKeys(customer.getFirstname());
        driver.findElement(By.name("lastname")).sendKeys(customer.getLastname());
        driver.findElement(By.name("address1")).sendKeys(customer.getAddress());
        driver.findElement(By.name("postcode")).sendKeys(customer.getPostcode());
        driver.findElement(By.name("city")).sendKeys(customer.getCity());

        driver.findElement(By.cssSelector("[id ^= select2-country_code]")).click();
        driver.findElement(By.cssSelector(
                String.format(".select2-results__option[id $= %s", customer.getCountry()))).click();
        wait.until((WebDriver d) -> d.findElement(
                By.cssSelector(String.format("select[name=zone_code] option[value=%s]", customer.getZone()))));
        new Select(driver.findElement(By.name("zone_code"))).selectByValue(customer.getZone());

        driver.findElement(By.name("email")).sendKeys(customer.getEmail());
        driver.findElement(By.name("phone")).sendKeys(customer.getPhone());
        driver.findElement(By.name("password")).sendKeys(customer.getPassword());
        driver.findElement(By.name("confirmed_password")).sendKeys(customer.getPassword());
        driver.findElement(By.name("create_account")).click();
    }

}

File: js-example/tests/customer_registration_tests.js
var target = require('../app/application'),
    test = require('selenium-webdriver/testing');

test.describe('Litecart', function() {
    var app;

    test.before(function() {
        app = new target.Application();
    });

    [{firstname: 'Adam', lastname: 'Smith', phone: '+0123456789',
      address: 'Hidden Place', postcode: '12345', city: 'New City',
      country: 'US', zone: 'KS',
      email: 'adam'+new Date().getTime()+'@smith.me',
      password: 'qwerty'}
    ].forEach(function(customer) {
        test.it('can register a customer', function() {
            var oldIds = app.getCustomerIds();

            app.registerNewCustomer(customer);

            var newIds = app.getCustomerIds();

            // сюда надо ещё добавить проверки
        });
    });

    test.after(function() {
        app.quit();
    });
});
